src/main/java/appeng/me/storage/CellInventory.java

```diff
@@ -218,11 +218,12 @@ public IAEItemStack injectItems( final IAEItemStack input, final Actionable mode
 			return input;
 		}
 
-		final ItemStack sharedItemStack = input.createItemStack();
-
-		if( CellInventory.isStorageCell( sharedItemStack ) )
+		// This is slightly hacky as it expects a read-only access, but fine for now.
+		// TODO: Guarantee a read-only access. E.g. provide an isEmpty() method and ensure CellInventory does not write
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)